demos/logging-stack/logging_stack.h

```diff
@@ -72,57 +72,118 @@
  * Common macro for all logging interface macros.
  */
 #if !defined( DISABLE_LOGGING )
+
+/**
+ * @brief Single common macro that maps all the logging interfaces,
+ * (#LogDebug, #LogInfo, #LogWarn, #LogError) to the platform-specific logging
+ * function.
+ *
+ * `printf` from the standard C library is the POSIX platform implementation used
+ * for logging functionality.
+ */
     #define SdkLog( string )    printf string
 #else
     #define SdkLog( string )
 #endif
 
+/* Disable definition of logging interface macros when generating doxygen output,
+ * to avoid conflict with documentation of macros at the end of the file. */
+#ifndef DOXYGEN
```

Comment:
I'm not sure about this `#ifndef DOXYGEN` and changing this code just for doxygen. Why don't we just have `@cond` here and `@endcond` at the end for doxygen to ignore this block? What do you think, @sarenameas ?

Reply:
I agree

Reply:
Updated to use `@cond` and `@endcond`